core/src/main/java/org/infinispan/interceptors/BaseCustomAsyncInterceptor.java

```diff
@@ -21,7 +21,7 @@
  * @author Dan Berindei
  * @since 9.0
  */
-public class BaseCustomSequentialInterceptor extends DDSequentialInterceptor {
+public class BaseCustomAsyncInterceptor extends DDAsyncInterceptor {
    protected Cache<?, ?> cache;
    protected EmbeddedCacheManager embeddedCacheManager;
 
```

Comment:
Nope, it tries to check whether the field `cache` was already initialized with a different cache instance. I think there's already enough info in the exception message below, so I won't add a comment.